main.tf

```diff
@@ -150,10 +174,11 @@ resource "azurerm_kubernetes_cluster" "main" {
     }
   }
   dynamic "microsoft_defender" {
-    for_each = var.microsoft_defender_enabled ? ["microsoft_defender"] : []
+
+    for_each = (var.microsoft_defender_enabled && var.log_analytics_workspace_enabled) ? ["microsoft_defender"] : []
```

Comment:
Since we have a `precondition` to prevent enable ms defender without enable log analytics workspace, could we simplify this line to the following snippet?:

```hcl
  for_each = var.microsoft_defender_enabled ? ["microsoft_defender"] : []
```

Reply:
Agreed.